network/topics/scoring.go

```diff
@@ -254,12 +255,16 @@ func filterCommitteesForTopic(topic string, committees []*storage.Committee) []*
 	return topicCommittees
 }
 
-// truncateStats returns the subnets in a small format topicNum=ti,fmd,mmd,imd
-func truncateStats(filtered map[string]*pubsub.TopicScoreSnapshot) []string {
-	filteredStrings := make([]string, 0, len(filtered))
+// formatInvalidMessageStats returns the subnets in a small format topicNum=ti,fmd,mmd,imd
+func formatInvalidMessageStats(filtered map[string]*pubsub.TopicScoreSnapshot) string {
+	var b strings.Builder
+	i := 0
 	for topic, snapshot := range filtered {
-		formatted := fmt.Sprintf("%s=%d;%0.3f;%0.3f;%0.3f", topic, snapshot.TimeInMesh, snapshot.FirstMessageDeliveries, snapshot.MeshMessageDeliveries, snapshot.InvalidMessageDeliveries)
-		filteredStrings = append(filteredStrings, formatted)
+		if i > 0 {
+			b.WriteString(" ")
+		}
+		fmt.Fprintf(&b, "%s=%d,%0.3f,%0.3f,%0.3f", topic, snapshot.TimeInMesh, snapshot.FirstMessageDeliveries, snapshot.MeshMessageDeliveries, snapshot.InvalidMessageDeliveries)
+		i++
 	}
-	return filteredStrings
+	return b.String()
```

Comment:
i forgot, this currently prints 3 decimals points even if the number is round, so that beats the purpose of us trying to reduce the total length

lets try this?

```go
// formatInvalidMessageStats returns the subnets in a small format topicNum=ti,fmd,mmd,imd
func formatInvalidMessageStats(filtered map[string]*pubsub.TopicScoreSnapshot) string {
	fmtFloat := func(n float64) string {
		if math.Trunc(n) == n {
			return strconv.FormatInt(int64(n), 10)
		}
	}
	var b strings.Builder
	i := 0
	for topic, snapshot := range filtered {
		if i > 0 {
			b.WriteString(" ")
		}
		fmt.Fprintf(&b, "%s=%d,%s,%s,%s", topic, fmtFloat(snapshot.TimeInMesh), fmtFloat(snapshot.FirstMessageDeliveries), fmtFloat(snapshot.MeshMessageDeliveries), fmtFloat(snapshot.InvalidMessageDeliveries))
		i++
	}
	return b.String()
}
```

this should print without decimals if the number is round or with 2 decimals if not

on another note i think numbers are always round so maybe we can even convert them to integers instead? @MatheusFranco99

Reply:
Great!!
TimeInMesh is round but the other counters suffer decay so it may really become real floats 😅 
I agree with adding this, good job!